internal/services/containerapps/helpers/container_apps.go

```diff
@@ -2257,21 +2288,36 @@ func SecretsDataSourceSchema() *pluginsdk.Schema {
 	}
 }
 
-func ExpandContainerSecrets(input []Secret) *[]containerapps.Secret {
+func ValidateContainerSecret(s Secret) error {
```

Comment:
This function shall not be exposed:
```suggestion
func validateContainerSecret(s Secret) error {
```

Reply:
TIL - I'm new to go, didn't realise that was how you specify private functions